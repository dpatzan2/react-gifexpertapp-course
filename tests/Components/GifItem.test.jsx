import { render, screen} from "@testing-library/react"
import { GifItem } from "../../src/Components/GifItem";

describe('Pruebas en GifItem', () => {

    const title = 'kokun';
    const url = 'https://github.com/'

    test('debe de hacer match con el snapshot', () => {
        const  {container}  = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar la imagen con el url y el ALt indicado', () => {
        render(<GifItem title={title} url={url} />);
        //screen.debug();
        //expect(screen.getByRole('img').src).toBe(url);

        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);
    });

    test('debe de mostrar el titulo en el componente', () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    });
})