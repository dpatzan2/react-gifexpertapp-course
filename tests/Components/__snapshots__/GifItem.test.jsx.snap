// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="kokun"
      src="https://github.com/"
    />
    <p>
      kokun
    </p>
  </div>
</div>
`;
